### PR TITLE
Remove fixed len constraint in FIPS mode

### PR DESCRIPTION
**Pull request by kaleb-himes**

### CODE REVIEW

**Comment by kaleb-himes** [comment]
David,

This has come up on enough FIPS projects now I'm opening a PR for it, part of CAVP testing is to make sure we can handle invalid lengths so we do not want strict length checking in FIPS mode.

- KH

File: IDE/GCC-ARM/Header/user_settings.h [hunk 1]
```diff
@@ -171,16 +171,15 @@ extern "C" {
...
     #define ECC_TIMING_RESISTANT
 
-    /* Enable cofactor support */
     #ifdef HAVE_FIPS
         #undef  HAVE_ECC_CDH
-        #define HAVE_ECC_CDH
-    #endif
+        #define HAVE_ECC_CDH /* Enable cofactor support */
+
+        #undef NO_STRICT_ECDSA_LEN
```
Comment (dgarske) [inline IDE/GCC-ARM/Header/user_settings.h:178]: In addition wouldn't it make sense to add something like this at the bottom of `wolfssl/wolfcrypt/settings.h`?

```
/* For FIPS make sure the ECDSA encoding allows extra bytes */
#if !defined(NO_STRICT_ECDSA_LEN) && defined(HAVE_FIPS) && \
        defined(HAVE_FIPS_VERSION) && HAVE_FIPS_VERSION >= 2    
    #define NO_STRICT_ECDSA_LEN
#endif
```

**Comment by dgarske** [comment]
For reference this issue traces back to https://github.com/wolfSSL/wolfssl/pull/2094

**Comment by toddouska** [comment]
> David,
> 
> This has come up on enough FIPS projects now I'm opening a PR for it, part of CAVP testing is to make sure we can handle invalid lengths so we do not want strict length checking in FIPS mode.
> 
> * KH

What do you mean by handle?  w/o NO_STRICT_ECDSA_LEN we handle bad sig lengths with an ASN_ECC_KEY_E.  Are you saying that in FIPS mode we have to not only allow bad sig lengths but treat them as correct?

**Comment by kaleb-himes** [comment]
@toddouska,

"handle invalid lengths" should have been written as "handle alternate lengths". We receive the raw elements from the FIPS lab, Qx, Qy, R, and S, import and verify them. By forcing a strict length some of the input vectors result in a "FAIL" when they should result in a "PASS".

**Pull request merged**